app/experimenter/experiments/admin/nimbus.py

```diff
@@ -22,6 +30,122 @@ def force_fetch_jetstream_data(modeladmin, request, queryset):
         tasks.fetch_experiment_data.delay(experiment.id)
 
 
+# Monkeypatch DecimalWidget render fn to work around a bug exporting Decimal to YAML
+# - https://github.com/django-import-export/django-import-export/issues/1324
+def DecimalWidgetRender(self, value, obj=None):
+    return str(value)
+
+
+DecimalWidget.render = DecimalWidgetRender
+
+
+class NimbusBranchForeignKeyWidget(ForeignKeyWidget):
+    """
+    override ForeignKeyWidget to filter NimbusBranch
+    by experiment in addition to the passed field
+    """
+
+    def __init__(self, model, field="slug", *args, **kwargs):
+        self.model = model
+        self.field = field
+
+    def get_queryset(self, value, row, *args, **kwargs):
+        experiment = NimbusExperiment.objects.get(slug=row.get("slug"))
+        return self.model.objects.filter(experiment=experiment)
+
+
+class NimbusExperimentResource(resources.ModelResource):
+    changes = fields.Field()
+    branches = fields.Field()
+    reference_branch_slug = fields.Field(
+        column_name="reference_branch_slug",
+        widget=NimbusBranchForeignKeyWidget(NimbusBranch, "slug"),
+    )
+
+    def get_diff_headers(self):
+        skip_list = ["reference_branch_slug"]
+        headers = []
+        for field in self.get_export_fields():
+            if force_text(field.column_name) in skip_list:
+                continue
+            headers.append(force_text(field.column_name))
+        return headers
+
+    class Meta:
+        model = NimbusExperiment
+        exclude = ("id", "reference_branch")
+        import_id_fields = ("slug",)
+
+    def dehydrate_changes(self, experiment):
+        all_changes = experiment.changes.all()
+
+        changelogs = []
+        for c in all_changes:
+            changelogs.append(dict(NimbusChangeLogSerializer(c).data))
```

Comment:
I'm not a huge fan of single letter variable names outside of comprehensions, but that's just me.